src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/SemanticTokensOrSemanticTokensEdits.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic.Models
+{
+    public struct SemanticTokensOrSemanticTokensEdits
```

Comment:
This is what [O# did](https://github.com/OmniSharp/csharp-language-server-protocol/blob/master/src/Protocol/Models/Proposals/SemanticTokensOrSemanticTokensEdits.cs).

The spec declares that `textdocument/semanticTokens/edits` should return `SemanticTokens | SemanticTokensEdits`, which is all fine and dandy in TypeScript, but a stranger proposition in C#. If you have ideas on a better way I'm game, this kind of class + the converter gives me the heebie jeebies.

Reply:
Ya i'm totally with you. Is the reason it can accept both because if the server can't resolve an "edit" it just returns all the tokens for the document?

Reply:
> Ya i'm totally with you. Is the reason it can accept both because if the server can't resolve an "edit" it just returns all the tokens for the document?

That's it. Should have been more explicit there, gotta remember that not everyone's been staring at that spec for days on end lol.

Reply:
I see. Ya it really blows to have that data type. Causes so much hassle 😢